### PR TITLE
Patch for foundry 0.8.X

### PR DESCRIPTION
**Pull request by p4535992**
I haven't implemented anything new I just updated the classes and objects from 0.7.X to 0.8.X. 
After some tests it seems to work correctly.
Good starting point for anyone who wants to actively develop new functions

### CODE REVIEW

File: README.md
Comment (Varriount) [inline README.md:1]: I'm not sure I'd put "Maintenance Mode" - that suggests that new features aren't welcome. The reality is that there isn't a dedicated maintainer at the moment.

File: .gitignore
Comment (Varriount) [inline .gitignore:1]:
```suggestion
.idea
node_modules
```

File: README.md
Comment (Varriount) [inline README.md:13]:
```suggestion
## NOTE: If you are a Javascript developer and not a Typescript developer, you can just use the Javascript files under the dist folder or rename a file from .ts to .js

```

File: dist/init.js
Comment (Varriount) [inline dist/init.js:40]: - At first glance, this doesn't look quite correct - I believe `log` is missing a conditional?
- The log levels should have (unexported) constants
- I would suggest converting the arrow functions into regular function declarations. In this instance, using arrow functions doesn't really provide any benefit.
- Unless they are really simple (`if (disabled) return;`) `if` statements should use braces and have correct indentation (in other words, don't try to fit them on as few lines as possible).
- `CONFIG.debug.hooks` should not normally be set by modules.

File: dist/lang/es.json
Comment (Varriount) [inline dist/lang/es.json:2]: Why the change to the module's localization key?

File: dist/lang/missing/ko.json
Comment (Varriount) [inline dist/lang/missing/ko.json:1]: Should these entries be blank? Does Foundry's localization mechanism fall back to English?

File: dist/templates/welcome-screen.html
Comment (Varriount) [inline dist/templates/welcome-screen.html:1]: How much effort would it take to get rid of this? Module welcome screens are, to me, generally just an annoyance.

File: foundryconfig.json
Comment (Varriount) [inline foundryconfig.json:2]: ?

**Review by Varriount — CHANGES_REQUESTED** [review]
- Should "prompt.js.bak" still exist?
- Please remove `package/forien-unidentified-items-v0.3.0.zip`
- Why were MJS files renamed with the Typescript file extension?
- Has ESLint been run over the files?
- It appears that Git has missed some files. I would appreciate if you could mark them as moved, by making all the `move` changes in a single commit. If you aren't sure how to do this, then just list the moved files and their destinations, and I can perform the rewrite and/or give you the right command.

I haven't reviewed everything yet (there's quite a lot of changes)

File: README.md
Comment (p4535992) [reply README.md:1]: No problem i removed the annotation

File: .gitignore
Comment (p4535992) [reply .gitignore:1]: No problem i will apply the changes

File: README.md
Comment (p4535992) [reply README.md:13]: No problem i will apply the changes

File: dist/init.js
Comment (p4535992) [reply dist/init.js:40]: `
- At first glance, this doesn't look quite correct - I believe log is missing a conditional? **(Sorry could you be clearer?)**
- The log levels should have (unexported) constants **(yea nothing wrong with that)**
- I would suggest converting the arrow functions into regular function declarations. In this instance, using arrow functions doesn't really provide any benefit. **(not necessary but i can do that too)**
- Unless they are really simple (if (disabled) return;) if statements should use braces and have correct indentation (in other words, don't try to fit them on as few lines as possible). **(sure not a problem)**
- CONFIG.debug.hooks should not normally be set by modules.  **(it is not entirely true some modules implement it for example "midi-qol" (from which I took inspiration for the [functions](https://gitlab.com/tposney/midi-qol/-/blob/master/src/midi-qol.ts)), but I can remove it without problems)**

`

So just to be clear (taking just the log function as an example) I should replace this function:

`export let log = (...args) => console.log(`${FORIEN_UNIDENTIFIED_ITEMS_MODULE_NAME} | `, ...args);`

with this ?

`export let log = function(...args){ console.log(`forien-unidentified-items | `, ...args) };`

File: dist/lang/es.json
Comment (p4535992) [reply dist/lang/es.json:2]: A "standard" that I have seen in other modules, that of using the module id as the primary key of the language files and that I also use, but it is a preference at the development level, I do not see particular problems, but I can restore the old label if you prefer it

File: dist/lang/missing/ko.json
Comment (p4535992) [reply dist/lang/missing/ko.json:1]: These files were present, even in the original project, but that's a good question, I don't think it does the fallback to the English language I would just remove them.

File: dist/templates/welcome-screen.html
Comment (p4535992) [reply dist/templates/welcome-screen.html:1]: No problem I remove it

File: foundryconfig.json
Comment (p4535992) [reply foundryconfig.json:2]: Just some dirty i forgot to put on the .gitignore. No problem I remove it

File: dist/lang/es.json
Comment (Varriount) [reply dist/lang/es.json:2]: No, this is fine. I was just curious.

**Comment by p4535992** [comment]
- Should "prompt.js.bak" still exist? **(Removed)**
- Please remove package/forien-unidentified-items-v0.3.0.zip  **(Removed)**
- Why were MJS files renamed with the Typescript file extension? **(the code is available in both typescript and javascript in order to be used by both types of developers, my departure is still from the typescript, but if someone wanted to take control of it, they would have no problems, i generate all the file with .js extension so if anyone wanted to use the files with the extension ".mjs" they would either have to rename them by hand, for now I don't plan to create an automated batch for that)**
- Has ESLint been run over the files? **(now eslint and prettier are applied on the code, details on the readme.md)**
- It appears that Git has missed some files. I would appreciate if you could mark them as moved, by making all the move changes in a single commit. If you aren't sure how to do this, then just list the moved files and their destinations, and I can perform the rewrite and/or give you the right command.**(I'm not sure how you want to handle it .. would you like me to reset the fork so that you can get it all in one commit?)**
- File .gitignore **(add .idea)**
- File readme.md **(updated)**
- File init.js  **(I have not fixed the log functions because they have never given me problems, but you are free to fix them as you see fit)**
- Removed all "missing" file localization files.
- File welcome-screen.html  **(Removed)**
- File prompt.js **(removed)**
- File foundryconfig.json **(I wanted to remove it, but then I left it as an example for those who want to use the "npm run-scrip't build" command as described in the readme.md file)**

File: .github/FUNDING.yml
Comment (mclemente) [inline .github/FUNDING.yml:6]: Hey, you got the changes from my fork, but you don't need to change this file since I meant to take the project over in case no one showed up to maintain it. I'd probably remove the file altogether unless someone else takes it over.

**Comment by p4535992** [comment]
Any update on this? If you don't like the typescript solution, I close the PR and pass the ball to clemente.

**Comment by Varriount** [comment]
> If you really want I can remove the dist folder from the indexes is the only thing that comes to mind to avoid confusion.

Yes, please do this. Generally speaking, it's not a good idea to add build-tool output to a repository, even if it does allow those who don't know typescript to read the code, because it creates the risk of people making PRs by modifying the `dist` files. The [release bundles](https://github.com/League-of-Foundry-Developers/foundryvtt-forien-unidentified-items/releases/tag/v0.2.6) will have the Javascript code anyway.

**Comment by Varriount** [comment]
Could you resolve the merge conflict? I believe it was caused by the recent merge of the Polish translation PR. Once that's done I can merge this PR in.

**Review by Varriount — APPROVED** [review]

File: src/modules/ContextMenu.ts
Comment (p4535992) [reply src/modules/ContextMenu.ts]: No problem i reformat the code

File: src/modules/ContextMenu.ts
Comment (p4535992) [reply src/modules/ContextMenu.ts]: No problem i reformat the code

File: src/modules/ContextMenu.ts
Comment (p4535992) [reply src/modules/ContextMenu.ts:76]: because for the typesafe based on 'foundry.js' it can be undefined "Game.items ?: Items | undefined"

**Comment by Varriount** [comment]
It looks like `package/forien-unidentified-items-v0.3.0.zip` snuck in, could you remove that?

**Comment by p4535992** [comment]
> 
> 
> It looks like `package/forien-unidentified-items-v0.3.0.zip` snuck in, could you remove that?
> Or if you're willing to wait, I can do it later this week.

i need to do the usually tests after the changes i made, should be all 99% fine anyway. I will do probably this weekend (and remove the zip too).

**Comment by p4535992** [comment]
Ok done the tests found some errors and corrected.
Let me know.

**Pull request merged**
